src/context/calculator/context.tsx

```diff
@@ -0,0 +1,47 @@
+import { useReducer, useContext, createContext, Dispatch, ReactNode } from 'react';
+
+import { OPERAND_INITIAL_VALUE, OPERATOR_INITIAL_VALUE } from '../../constants/calculator';
+import { InitialOperator, Operator } from '../../types/calculator';
+import { Action } from './action';
+import { calculatorReducer } from './reducer';
+
+export const initialState = {
+  operand1: OPERAND_INITIAL_VALUE,
+  operand2: OPERAND_INITIAL_VALUE,
+  operator: OPERATOR_INITIAL_VALUE,
+} as const;
+
+export type State = {
+  operand1: number;
+  operand2: number;
+  operator: Operator | InitialOperator;
+};
```

Comment:
하나의 파일 내에 타입과 구현체를 선언해야 한다면, 타입을 위로 두시는 건 어떨까요? `initialState`가 `State`에 대한 구현체이기 때문에 타입을 명시적으로 지정해줌으로써 타입 안정성을 확보할 수 있을 것 같습니다.

```suggestion
export type State = {
  operand1: number;
  operand2: number;
  operator: Operator | InitialOperator;
}; 

export const initialState: State = {
  operand1: OPERAND_INITIAL_VALUE,
  operand2: OPERAND_INITIAL_VALUE,
  operator: OPERATOR_INITIAL_VALUE,
} as const; 
```

Reply:
저도 타입을 먼저 선언하고 타입에 대한 구현을 하는 흐름이 알맞다고 생각합니다.
이러한 경우에는 명시적으로 타입을 지정해주는 것도 좋은 것 같네요!